components/browser/icons/src/test/java/mozilla/components/browser/icons/processor/DiskIconProcessorTest.kt

```diff
@@ -83,4 +84,40 @@ class DiskIconProcessorTest {
 
         verify(cache, never()).put(any(), any(), any(), eq(icon))
     }
+
+    @Test
+    fun `Icon loaded in private mode is not saved in cache`() {
+        val icon: Icon = mock()
+
+        /* Can be Source.INLINE as well. To ensure that the icon is eligible for caching on the disk. */
+        `when`(icon.source).thenReturn(Icon.Source.DOWNLOAD)
```

Comment:
This is kinda ugly, but was the only option. Ideally, if Mockito allowed mocking static variables, then we could have mocked Icon.shouldCacheOnDisk extension property.

Reply:
Instead of mocking `Icon` you could just mock `Bitmap` so you don't have to deal with the getters.

```kt
val icon = Icon(mock, source = Icon.Source.DOWNLOAD)
```

Reply:
@mauryat Do you want to try that? Other than that the PR looks good to me.